test/integration/cartridge_health_test.lua

```diff
@@ -1,40 +1,78 @@
-local fio = require('fio')
-local t = require('luatest')
+local fio = require("fio")
+local t = require("luatest")
 local g = t.group()
 
-local helpers = require('test.helper')
+local helpers = require("test.helper")
 
-g.before_all(function()
-    t.skip_if(type(helpers) ~= 'table', 'Skip cartridge test')
-    g.cluster = helpers.init_cluster()
-end)
+g.before_all(
+    function()
+        t.skip_if(type(helpers) ~= "table", "Skip cartridge test")
+        g.cluster = helpers.init_cluster()
+    end
+)
 
-g.after_all(function()
-    g.cluster:stop()
-    fio.rmtree(g.cluster.datadir)
-end)
+g.after_all(
+    function()
+        g.cluster:stop()
+        fio.rmtree(g.cluster.datadir)
+    end
+)
 
 g.test_cartridge_health_handler = function()
-    helpers.skip_cartridge_version_less('2.0.2')
+    helpers.skip_cartridge_version_less("2.0.2")
     helpers.upload_default_metrics_config(g.cluster)
-    local main_server = g.cluster:server('main')
-    local resp = main_server:http_request('get', '/health', {raise = false})
+    local main_server = g.cluster:server("main")
+    local resp = main_server:http_request("get", "/health", {raise = false})
     t.assert_equals(resp.status, 200)
 end
 
 g.test_cartridge_health_fail_handler = function()
-    helpers.skip_cartridge_version_less('2.0.2')
+    helpers.skip_cartridge_version_less("2.0.2")
     helpers.upload_default_metrics_config(g.cluster)
-    local main_server = g.cluster:server('main')
-    main_server.net_box:eval([[
+    local main_server = g.cluster:server("main")
+    main_server.net_box:eval(
+        [[
         _G.old_info = box.info
         box.info = {
             status = 'orphan',
         }
-    ]])
-    local resp = main_server:http_request('get', '/health', {raise = false})
+    ]]
+    )
+    local resp = main_server:http_request("get", "/health", {raise = false})
     t.assert_equals(resp.status, 500)
     main_server.net_box:eval([[
         box.info = _G.old_info
     ]])
 end
+
+g.test_cartridge_health_handler_member_state_configured_configuring = function()
+    helpers.skip_cartridge_version_less("2.0.2")
+    helpers.upload_default_metrics_config(g.cluster)
+    local main_server = g.cluster:server("main")
+    main_server.net_box:eval(
+        [[
+        myself = require('membership').myself()
+
+        myself.payload.state='ConfiguringRoles'
+        myself.payload.state_prev='RolesConfigured'
+    ]]
+    )
+    local resp = main_server:http_request("get", "/health", {raise = false})
+    t.assert_equals(resp.status, 200)
+end
+
+g.test_cartridge_health_handler_member_state_boxconfigured_configuring = function()
+    helpers.skip_cartridge_version_less("2.0.2")
+    helpers.upload_default_metrics_config(g.cluster)
+    local main_server = g.cluster:server("main")
+    main_server.net_box:eval(
+        [[
+        myself = require('membership').myself()
```

Comment:
I think it won't work. You change fields of a global variable `myself` instead of `membership.myself()` function. 
You need smth like this:
```lua
package.loaded.membership.myself = function()
    return {
        payload = {
              state='ConfiguringRoles',
              state_prev='BoxConfigured',
       }
    }
end
```

Reply:
And please provide a screenshot of the tests run

Reply:
But it worked. I am not very familiar with luatest and TT internals, thanks for advice. I will try..

test/integration

![Screenshot 2022-10-21 at 13 30 32](https://user-images.githubusercontent.com/17274766/197196209-bbffb64c-065b-47da-b244-320914643ad8.png)

test/integration/cartridge_health_test.lua

![Screenshot 2022-10-21 at 13 30 45](https://user-images.githubusercontent.com/17274766/197196158-06fdb47c-2de6-43f9-b9ca-8104791c3365.png)

Reply:
Updated.

![Screenshot 2022-10-21 at 14 25 01](https://user-images.githubusercontent.com/17274766/197206480-c40d045d-376f-42ab-a80d-0f53e7c116d8.png)